builtin/providers/aws/resource_aws_ssm_document.go

```diff
@@ -0,0 +1,309 @@
+		Update: resourceAwsSsmDocumentUpdate,
+		Delete: resourceAwsSsmDocumentDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"content": &schema.Schema{
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"created_date": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hash": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hash_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"owner": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"platform_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"parameter": &schema.Schema{
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"default_value": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"description": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"type": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"permissions": &schema.Schema{
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"account_ids": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsSsmDocumentCreate(d *schema.ResourceData, meta interface{}) error {
+	ssmconn := meta.(*AWSClient).ssmconn
+
+	log.Printf("[INFO] Creating SSM Document: %s", d.Get("name").(string))
+
+	docInput := &ssm.CreateDocumentInput{
+		Name:    aws.String(d.Get("name").(string)),
+		Content: aws.String(d.Get("content").(string)),
+	}
+
+	resp, err := ssmconn.CreateDocument(docInput)
+
+	if err != nil {
+		return fmt.Errorf("[ERROR] Error creating SSM document: %s", err)
+	}
+
+	d.SetId(*resp.DocumentDescription.Name)
+
+	if v, ok := d.GetOk("permissions"); ok && v != nil {
+		setDocumentPermissions(d, meta)
+	} else {
+		log.Printf("[DEBUG] not setting document permissions")
+	}
+
+	return resourceAwsSsmDocumentRead(d, meta)
+}
+
+func resourceAwsSsmDocumentRead(d *schema.ResourceData, meta interface{}) error {
+	ssmconn := meta.(*AWSClient).ssmconn
+
+	log.Printf("[DEBUG] Reading SSM Document: %s", d.Get("name").(string))
```

Comment:
I guess not. `name` just happens to be only unique element mentioned in the docs:
https://docs.aws.amazon.com/ssm/latest/APIReference/API_CreateDocument.html